packages/merge-on-green/src/merge-logic.ts

```diff
@@ -137,152 +149,115 @@ mergeOnGreen.hasMOGLabel = async function hasMOGLabel(
+ * @param owner of pr (from Watch PR)
+ * @param repo of pr (from Watch PR)
+ * @param github unique installation id for each function
+ * @returns a string array of the names of the required checks
+ */
 mergeOnGreen.getBranchProtection = async function getBranchProtection(
   owner: string,
   repo: string,
   github: GitHubAPI
-) {
-  const branchProtection = (
-    await github.repos.getBranchProtection({
-      owner,
-      repo,
-      branch: 'master',
-    })
-  ).data.required_status_checks.contexts;
-  console.log(`branch protection for ${owner}/${repo}: ${branchProtection}`);
-  return branchProtection;
-};
-
-mergeOnGreen.getRequiredChecks = async function getRequiredChecks(
-  github: GitHubAPI,
-  owner: string,
-  repo: string
 ): Promise<string[]> {
-  const [checksByLanguage, languageMap] = await Promise.all([
-    mergeOnGreen.requiredChecksByLanguage(github),
-    mergeOnGreen.mapReposToLanguage(github),
-  ]);
-  if (checksByLanguage && languageMap) {
-    const language = languageMap.find(
-      (element: Language) => element.repo === `${owner}/${repo}`
-    );
-    if (language !== undefined) {
-      if (checksByLanguage[language.language].repoOverrides !== undefined) {
-        const isOverriden = checksByLanguage[
-          language.language
-        ].repoOverrides.find(
-          (element: RepoOverrides) => element.repo === `${owner}/${repo}`
-        );
-        if (isOverriden) {
-          console.log(
-            `Your language's required checks were overridden because of the repo ${owner}/${repo}` +
-              `it is ${isOverriden.useBranchProtectionRules} that ${owner}/${repo} uses default branch protection`
-          );
-          if (isOverriden.useBranchProtectionRules === true) {
-            console.log(`Checking branch protection for ${owner}/${repo}`);
-            const branchProtection = await mergeOnGreen.getBranchProtection(
-              owner,
-              repo,
-              github
-            );
-            console.log(`Using native branch protection for ${owner}/${repo}`);
-            return branchProtection;
-          }
-          return isOverriden.requiredStatusChecks;
-        }
-      }
-      return checksByLanguage[language.language].requiredStatusChecks;
-    } else {
-      console.info(
-        `${owner}/${repo} does not have a corresponding language in sloth/repos.json`
-      );
-      return [];
-    }
-  } else {
-    console.info(
-      `Could not find any checks or a language map for ${owner}/${repo}`
+  try {
+    const branchProtection = (
+      await github.repos.getBranchProtection({
+        owner,
+        repo,
+        branch: 'master',
+      })
+    ).data.required_status_checks.contexts;
+    console.log(
+      `checking branch protection for ${owner}/${repo}: ${branchProtection}`
     );
+    return branchProtection;
+  } catch (err) {
+    console.log(`Error in getting branch protection: ${err}`);
```

Comment:
we should be mindful about continuing to add context about what repo is having this error. Eventually we should pull this logic up a layer, so that any bot just gets it for free if they use a built in logger.